clar2wasm/src/wasm_utils.rs

```diff
@@ -1284,19 +1285,7 @@ pub fn call_function<'a>(
 
     // Call the function
     func.call(&mut store, &wasm_args, &mut results)
-        .map_err(|e| {
-            // TODO: If the root cause is a clarity error, we should be able to return that,
-            //       but it is not cloneable, so we can't return it directly.
-            //       If the root cause is a trap from our Wasm code, then we need to translate
-            //       it into a Clarity error.
-            //       See issue stacks-network/clarity-wasm#104
-            // if let Some(vm_error) = e.root_cause().downcast_ref::<crate::vm::errors::Error>() {
-            //     vm_error.clone()
-            // } else {
-            //     Error::Wasm(WasmError::Runtime(e))
-            // }
-            Error::Wasm(WasmError::Runtime(e))
-        })?;
+        .map_err(|_| error_mapping::map(instance, &mut store))?;
```

Comment:
Does it actually satisfy the TODO that was here before?

Reply:
It seems fine, don't you think?
1. For Clarity errors (ie.: `ShortReturnType`, compile-time errors) we have aligned to handle them as other issues.
2. Wasm runtime errors are being handled within this PR.

Reply:
Not sure I understand what you mean. From what I understand from the TODO, we should check if the error is already an instance of clarity `Error`, and if not throw a runtime error.

Here, it seems we just ignore whatever is returned and go read the `runtime-error-code`. But it might fail for other reasons, no?

For example, let's imagine the function calls a host function that needs to read a value from the memory but fails to access it. `e` would be an error reflecting this, but we discard it. And this failure wouldn't change the `runtime-error-code`.

Reply:
I have changed the code to handle also Clarity `Error`.